### PR TITLE
Adding CONTRIBUTING.md content

### PR DESCRIPTION
**Pull request by samgnaniah**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+Building Ballerina requires a Java SE Development Kit (JDK) version 8 to be installed. You can download one from one of the following locations:
+
+* [http://www.oracle.com/technetwork/java/javase/downloads/index.html](http://www.oracle.com/technetwork/java/javase/downloads/index.html)
+* [http://openjdk.java.net/install/index.html](http://openjdk.java.net/install/index.html)
+
+> NOTE: Set an environment variable `JAVA_HOME` to the path name of the directory into which you installed the JDK release.
+
+**Prerequisite 2** - Install Maven 3.3+ or later 
+
+Install maven 3.3 or a later version: [https://maven.apache.org/install.html](https://maven.apache.org/install.html)
```
Comment (pubudu91) [inline CONTRIBUTING.md]: The build doesn't work on Maven 3.3. The dev team switched to 3.5.3, which is the latest.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+If you download the sources, you need to update the git submodules using the following command.
+
+```
+git submodule update --init 
+```
+
+Run the Maven command `mvn clean install` from the Ballerina root directory:
+
+Command | Description
```
Comment (pubudu91) [inline CONTRIBUTING.md]: Should we add `mvn clean install -P ballerina` to this list? The `ballerina` profile is used to build only the modules necessary for the runtime distribution (i.e., sans the tools etc.)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+```
+
+Run the Maven command `mvn clean install` from the Ballerina root directory:
+
+Command | Description 
+--- | ---
+`mvn clean install` | Build and install the artifacts into the local repository.
+`mvn clean install -Dmaven.test.skip=true` | Build and install the artifacts into the local repository, without running any of the unit tests.
+
+Extract the Ballerina distribution created at `distribution/zip/ballerina/target/ballerina-<version>-SNAPSHOT.zip`.
```
Comment (pubudu91) [inline CONTRIBUTING.md]: This only considers the runtime distribution. Shouldn't we be describing both, the runtime distribution and the platform distribution?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+* BallerinaParser.g4 - Contains the parser rules. Parser listens to the token stream generated by the lexer. High level grammar productions/abstractions are defined in the parser using those tokens.
+
+> TIP: If you want to check and validate a grammar rule you just wrote, you can use the antlr4 plugin of your IDE. See Setting up your development environment section for more details.
+
+#### Generating Parsers
+
+Once a change is done to the any of the grammar files, it is needed to re-generate the parsers. To generate parsers, we first need to download antlr-complete-4.5.3.jar. Then navigate to <ballerina>/docs/grammar, and generate the lexer followed by the parser (order is important) using the following command:
+
+```
+java -jar /home/supun/Downloads/antlr-4.5.3-complete.jar *.g4 -package org.wso2.ballerinalang.compiler.parser.antlr4 -o ../../compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/antlr4/.
```
Comment (pubudu91) [inline CONTRIBUTING.md]: Shall we generify the following path: `/home/supun/Downloads/antlr-4.5.3-complete.jar`?

Comment (pubudu91) [reply CONTRIBUTING.md]: For example, `<path_to_antlr_jar>/antlr-4.5.3-complete.jar`

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+##### Useful IDEA Plugins
+
+* ANTLR 4 - https://plugins.jetbrains.com/plugin/7358-antlr-v4-grammar-plugin 
+* Ballerina - https://plugins.jetbrains.com/plugin/9520-ballerina (Need IDEA 2016.3 or newer)
+
+#### Setup Eclipse
+
+##### Importing a Ballerina Project
+
+All Ballerina repositories are developed as maven projects. So you can import any Ballerina project to eclipse similar to any maven project.
```
Comment (pubudu91) [inline CONTRIBUTING.md]: eclipse -> Eclipse in all the places

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+**Prerequisite 1** - Install a Java Development Kit (JDK) version 8
+
+Building Ballerina requires a Java SE Development Kit (JDK) version 8 to be installed. You can download one from one of the following locations:
+
+* [http://www.oracle.com/technetwork/java/javase/downloads/index.html](http://www.oracle.com/technetwork/java/javase/downloads/index.html)
+* [http://openjdk.java.net/install/index.html](http://openjdk.java.net/install/index.html)
+
+> NOTE: Set an environment variable `JAVA_HOME` to the path name of the directory into which you installed the JDK release.
+
+**Prerequisite 2** - Install Maven 3.3+ or later
```
Comment (pubudu91) [inline CONTRIBUTING.md]: Version is still 3.3 here

Comment (samgnaniah) [reply CONTRIBUTING.md]: What should the version be?

File: CONTRIBUTING.md (hunk 3 above)
Comment (samgnaniah) [reply CONTRIBUTING.md]: Not sure whether we need platform distribution here. Is it relevant? If so what are the steps we need to add?

File: CONTRIBUTING.md (hunk 2 above)
Comment (samgnaniah) [reply CONTRIBUTING.md]: Will add.

Comment (samgnaniah) [reply CONTRIBUTING.md]: Added.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+
+Run the Maven command `mvn clean install` from the Ballerina root directory:
+
+Command | Description 
+--- | ---
+`mvn clean install` | Build and install the artifacts into the local repository.
+`mvn clean install -Dmaven.test.skip=true` | Build and install the artifacts into the local repository, without running any of the unit tests.
+`mvn clean install -P ballerina` | The `ballerina` profile is used to build only the modules necessary for the runtime distribution (i.e., excluding the tools, etc.)
+
+Extract the Ballerina distribution created at `distribution/zip/ballerina/target/ballerina-<version>-SNAPSHOT.zip`.
```
Comment (pubudu91) [inline CONTRIBUTING.md]: Need to be explicit about this. `zip/ballerina` contains the runtime only. `zip/ballerina-tools/` contains the runtime + tools (e.g., Composer)

**Pull request merged**